db-service/lib/cqn2sql.js

```diff
@@ -701,11 +701,36 @@ class CQN2SQLRenderer {
    * @returns {string} The correct operator string
    */
   operator(x, i, xpr) {
-    if (x === '=' && xpr[i + 1]?.val === null) return 'is'
-    if (x === '!=') return 'is not'
+    if (x === '=') return this.operator_has_null(i, xpr) ? 'is' : '='
+    if (x === '!=') return this.operator_has_null(i, xpr) ? 'is not' : '!='
```

Comment:
That only cares for converting invalid native SQL expressions into valid `is null` / `is not null` expressions. 
For example:

```sql
SELECT 1 where 1 = null;
SELECT 1 where 1 != null;
```
→ 
```sql
SELECT 1 where 1 is null;
SELECT 1 where 1 is not null;
```

Note that this is plain old ANSI SQL, not to be confused with `is [not] distinct from` operators.

Reply:
I have adjusted the logic to be ANSI SQL compliant. As far as I could find `IS NULL` and `IS NOT NULL` are the ANSI defined NULL operators. Where before the logic did not check the right hand side for the `!=` operator. Now it makes sure that the right hand side is always `{val:null}` to ensure that it would generate valid ANSI SQL.